doc/rfcs/0001-separate-component-version-in-cluster.md

```diff
@@ -0,0 +1,44 @@
+# Separate Component Version in Cluster
+
+## Summary
+
+Add a version field to each component on topology file. Allow user to upgrade single component to given version.
+
+## Motivation
+
+- New component TiDB Dashboard need to deployed with TiDB but not release with TiDB. Need a way to specify Dashboard version when deployment and upgrade.
+- User want to upgrade node_exporter without upgrade TiDB cluster
+- Maybe it could replace tiup cluster patch function to provide patch version of single component
+
+## Detailed design
+
+1. add "latest" alias to tiup download function.It cloud be used to download latest release package of component.
+
+2. add version field to DashboardSpec struct.
+
+```
+type DashboardSpec struct {
+	...
+	Version    string    `yaml:"version,omitempty" default:"latest"`
+	...
+}
+```
+
+3. add node_exporter and blackbox_exporter version to MonitoredOptions struct
+
+```
+MonitoredOptions struct {
+    ...
+	NodeExporterVersion     string    `yaml:"node_exporter_version,omitempty" default:"latest"`
+	BlackboxExporterVersion string    `yaml:"blackbox_exporter_version,omitempty" default:"latest"`
+    ...
+	}
+```
+
+4. if user not specify version, those version has default value latest which means use newest stable version in tiup repo.
+
+5. maybe add version field to all other component, deafult value is empty which means use global cluster version
```

Comment:
As comments above, we should only add component specified version fields to those claims to work with other components on different versions.

Reply:
Maybe some version constraint is also needed. For example, TiDB Dashboard vX.X requires TiDB >= vX.X.

Reply:
> Maybe some version constraint is also needed. For example, TiDB Dashboard vX.X requires TiDB >= vX.X and certain constraint break should not be allowed.

maybe we do not need it? if latest tidb-dashboard always support all exist tidb (>=6.2.0). just leave "latest" alisas as tidb-dashboard version. If user upgrade tidb cluster, tidb-dashboard upgrade to latest.If user do not upgrade tidb cluster, tidb-dashboard remains old version. We cloud just to warning user not to upgrade single component if he don't know what will happen.

Reply:
it is easy to add something like new dashboard do not support old tidb, but not easy to add information like old dashboard do not support new tidb